salvadorTests/FetchImage_Tests.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  FetchImage_Tests.swift
+//  salvadorTests
+//
+//  Created by Geury Roustand on 22.02.23.
+//
+
+import XCTest
+
+@testable import salvador
+
+final class FetchImage_Tests: XCTestCase {
+    
+    var createImage: CreateImage!
+    var viewModel : GenerateImageViewModel!
+    
+    @MainActor override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        createImage = CreateImage()
+        viewModel = GenerateImageViewModel()
+    }
+    
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        createImage = nil
+        viewModel = nil
+    }
+    
+    func testGenerateImage_SuccessFulResponsw() async throws {
```

Comment:
There is a small typo in `Response`.

My suggestion for a unit test naming pattern is the given-when-then pattern (here an `_` is actually useful). it would look like this:

```
func test_givenValidPromt_andValidAPIKey_whenSuccessfullyGenerateImage_thenReturnedDataResultIsNotEmpty() async throws {

    // given (preparing the prerequisites)

     let prompt = "Create a mango eating an apple with sunglases on the beach"
     let apiKey = await viewModel.loadJson(fileName: "APISecret")

    // when (triggering the function to test)

    let result = try await createImage.generateImage(withPrompt: prompt, apiKey: apiKey.APISecret)

    // then (making assertions on the result)

    XCTAssertNotNil(result.data)
    XCTAssertEqual(result.data.count, 1)
    XCTAssertNotNil(result.data[0].url)
}
```

It can get quite long, but it perfectly describes what happens. The "when" part can be optional but the "given" and "then" should be there at least.